### PR TITLE
go.mod: update jwt libraries

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Follow up for https://github.com/upbound/universal-crossplane/pull/241 . I had to make a bit more intrusive changes.

We use `github.com/nats-io/jwt/v2` now instead of `github.com/nats-io/jwt` and I needed to update `github.com/labstack/echo/v4` to get rid of `github.com/dgrijalva/jwt-go` because `replace with github.com/golang-jwt/jwt` wouldn't work with `go: github.com/golang-jwt/jwt@v3.2.1+incompatible used for two different module paths (github.com/dgrijalva/jwt-go and github.com/golang-jwt/jwt)`

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

`make reviewable` and push validation of the PR.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,16 +9,16 @@ require (
...
 	github.com/google/go-cmp v0.5.5
 	github.com/google/uuid v1.1.2
 	github.com/jarcoal/httpmock v1.0.8
 	github.com/labstack/echo-contrib v0.9.0
-	github.com/labstack/echo/v4 v4.2.2
-	github.com/labstack/gommon v0.3.0
-	github.com/nats-io/jwt v1.2.2
+	github.com/labstack/echo/v4 v4.7.2
+	github.com/labstack/gommon v0.3.1
+	github.com/nats-io/jwt/v2 v2.2.1-0.20220113022732-58e87895b296
```
[the remark this answers is not in the document]
Comment (hasheddan) [reply go.mod:21]: @muvaf is there a transitive dependency causing it?

[the remark this answers is not in the document]
Comment (muvaf) [reply go.mod:21]: I don't see one. It's `go mod tidy` that repeatedly changes it from `v2.2.0` to this tag. I couldn't really figure out why but probably related to tagging and Github's go module server.

[the remark this answers is not in the document]
Comment (hasheddan) [reply go.mod:21]: Interesting 🤔 it appears `v2.2.0` is a few commits behind (https://github.com/nats-io/jwt/commits/master) -- I don't think we need to block on this, but would like to understand the behavior

**Review by hasheddan — APPROVED** [review]

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (muvaf) [reply go.mod:21]: Honestly, gomod behavior with Github tags is always confusing for me. Not sure how to debug this specific behavior.

**Pull request merged**